plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFieldsFrontend.php

```diff
@@ -185,23 +185,33 @@ public function edit_account_form_fields() {
 	 *
 	 * @param integer $user_id User ID.
 	 */
-	public function save_account_form_fields( $user_id ) {
-		// phpcs:disable WordPress.Security.NonceVerification.Missing
-		$customer          = new WC_Customer( $user_id );
+	public function save_account_form_fields( int $user_id ) {
+
+		try {
+			$customer = new WC_Customer( $user_id );
+		} catch ( \Exception $e ) {
+			wc_add_notice(
+				__( 'Unable to save customer additional fields: ', 'woocommerce' ) . $e->getMessage(),
+				'notice'
+			);
+			return;
+		}
+
 		$additional_fields = $this->checkout_fields_controller->get_fields_for_location( 'contact' );
-		$field_values      = array();
 
-		foreach ( array_keys( $additional_fields ) as $key ) {
-			$post_key = CheckoutFields::get_group_key( 'other' ) . $key;
```

Comment:
Bug source was this key. The field name on the form isn't prefixed with the group.